packages/request-node/src/thegraph-node.ts

```diff
@@ -25,12 +28,17 @@ export class TheGraphRequestNode extends RequestNodeBase {
     );
     const signer = new NonceManager(wallet);
     const ipfsStorage = getIpfsStorage(logger);
+    const storage = new EthereumStorageEthers({
+      ipfsStorage,
+      signer,
+      network,
+      logger,
+    });
     const dataAccess = new TheGraphDataAccess({
       graphql: { url },
-      ipfsStorage,
+      storage,
       network,
-      signer,
-      logger: logger,
+      logger,
     });
 
     super(dataAccess, ipfsStorage, store, logger);
```

Comment:
Have you kept `ipfsStorage` on purpose? It looks weird so maybe add a comment to explain

Reply:
@benjlevesque I understand what @yomarion means (I think). I agree that it's weird to have an `ipfsStorage` object in the `RequestNodeBase` constructor, knowing that we pass down a `dataAccess` object that should already be a layer on top of IPFS. The issue is that it doesn't look possible to abstract—in the data-access layer—what `ipfsStorage` is used for in the request-node: the `/ipfsAdd` endpoint. This endpoint only persists documents on IPFS without persisting anything on-chain. I don't think it makes sense to add a generic method in data-access that only persist documents off-chain (like `persistTransactionOffChain` for instance). The data-access interface shouldn't contain methods that are tied to the underlying technology used for persisting IMO. Honestly, I think this endpoint is a security hole anyway, allowing users to persist documents on IPFS without persisting anything on-chain means we disregard spamming protection and the IPFS node could quickly become filled with fake documents. I'd propose we completely remove that endpoint, I don't think it is used anywhere, right?

Reply:
Yeah sorry we solved this in DM, my bad. 

this endpoint should be challenged for sure. It is related to the option I removed from the client. I believe it _should_ be possible to use 100% the client as a front-end library, but it doesn't need to be first-class like it is today

Reply:
Ha! Get it, thanks!